### PR TITLE
Changed frankie_v1 Project To Actually Match The Prototype Board

### PR DESCRIPTION
**Pull request by garethellis0**
### Description
- updated the `frankie_v1` project to actually match the prototype board we're using rather then just the raw stm32h7 MCU
- removed RTOS support from the project since we're not using it yet, and simpler makes it easier for elec to iterate faster
- changed the build script to forcibly link all symbols in so that the interrupts can actually fire
- the only thing I _really_ want review on here is the `BUILD` file

### Testing Done
- flashed a `stm32h743ZI` demo board with the firmware and ensure that the `systick` interrupt is called and we can debug

### Length Justification
- mostly generated code changes. not worth much review for now, we're going to be seeing a lot of changes like this as we prototype firmware for elec stuff

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/firmware_new/boards/frankie_v1/STM32H743ZITx_FLASH.ld [hunk 1]
```diff
@@ -55,8 +55,8 @@ ENTRY(Reset_Handler)
 /* Highest address of the user mode stack */
 _estack = 0x20020000;    /* end of RAM */
 /* Generate a link error if heap and stack don't fit into RAM */
-_Min_Heap_Size = 0x1000;      /* required amount of heap  */
-_Min_Stack_Size = 0x2000; /* required amount of stack */
+_Min_Heap_Size = 0x200;      /* required amount of heap  */
+_Min_Stack_Size = 0x400; /* required amount of stack */
```
[the remark this answers is not in the document]
Comment (garethellis0) [reply src/firmware_new/boards/frankie_v1/STM32H743ZITx_FLASH.ld:59]: They were increased for RTOS previously I believe. I basically reverted all settings to the defaults, so we have a clean(-ish) slate to start from.

**Review by MathewMacDougall — COMMENTED** [review]
Looks like we will want to ignore formatting in the new generated code since that won't pass CI.

**Comment by garethellis0** [comment]
I'd rather not ignore formatting, so fixed.

**Review by MathewMacDougall — APPROVED** [review]

File: src/firmware_new/boards/frankie_v1/stm32h7xx_hal_conf.h [hunk 2]
```diff
@@ -98,9 +98,9 @@ extern "C"
  */
 #if !defined(HSE_VALUE)
 #define HSE_VALUE                                                                        \
-    ((uint32_t)25000000) /*!< Value of the External oscillator in Hz : FPGA case fixed   \
-                            to 60MHZ */
-#endif                   /* HSE_VALUE */
+    ((uint32_t)8000000) /*!< Value of the External oscillator in Hz : FPGA case fixed to \
+                           60MHZ */
```
Comment (jonathanlew) [inline src/firmware_new/boards/frankie_v1/stm32h7xx_hal_conf.h:102]: Is this still 60MHz?

**Review by jonathanlew — COMMENTED** [review]
One itty bitty comment

File: src/firmware_new/boards/frankie_v1/stm32h7xx_hal_conf.h (hunk 2 above)
Comment (garethellis0) [reply src/firmware_new/boards/frankie_v1/stm32h7xx_hal_conf.h:102]: Yep, this is generated code, so even if it's wrong, not much we can do about it :grimacing:

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Pull request merged**